### PR TITLE
Typescript in builder

### PR DESCRIPTION
**Pull request by adrinr**
## Description
Enable ts files (instead of js) in the builder. This PR sets the config for it, and transforms both navigation and plugin store to typescript. Also converted the API file to ts, as it's consumed by these stores. All the other usages (that are consuming it from js) keep working, so this proves that the hybrid model is supported, consuming ts from js. This will allow a phased transformation to ts.

## Launchcontrol
Enabling typescript files in the builder.

## Feature branch env
[Feature Branch Link](http://fb-budi-8866-builder-enable-typescript.fb.qa.budibase.net)

### CODE REVIEW

File: packages/builder/.gitignore [hunk 1]
```diff
@@ -5,4 +5,5 @@ package-lock.json
 release/
 dist/
 routify
-.routify/
\ No newline at end of file
+.routify/
+.rollup.cache
```
Comment (samwho) [inline packages/builder/.gitignore:9]: How come this is needed?

**Review by samwho — APPROVED** [review]

File: packages/builder/.gitignore (hunk 1 above)
Comment (adrinr) [reply packages/builder/.gitignore:9]: Rollup is now building the ts files, and it creates this cache folder with a lot of bundled files. We don't want this to be in the repo

Comment (adrinr) [reply packages/builder/.gitignore:9]: image

**Review by mike12345567 — APPROVED** [review]
LGTM - very nice work!

File: packages/builder/vite.config.mjs [hunk 2]
```diff
@@ -3,6 +3,7 @@ import replace from "@rollup/plugin-replace"
 import { defineConfig, loadEnv } from "vite"
 import { viteStaticCopy } from "vite-plugin-static-copy"
 import path from "path"
+import typescript from "@rollup/plugin-typescript"
```
[the remark this answers is not in the document]
Comment (adrinr) [reply packages/builder/vite.config.mjs:6]: I think it's required due to some issues with svelte... Without this we get the following error:
image

**Pull request merged**